src/domain/core/src/services/push_service.rs

```diff
@@ -122,3 +128,36 @@ impl From<GetDatasetError> for PushError {
         }
     }
 }
+
+#[derive(Debug, Error)]
+pub enum StatusCheckError {
+    #[error(transparent)]
+    CompareChainsError(
+        #[from]
+        #[backtrace]
+        chain_cmp::CompareChainsError,
+    ),
+    #[error(transparent)]
+    BuildDatasetError(
+        #[from]
+        #[backtrace]
+        BuildDatasetError,
+    ),
+    #[error(transparent)]
+    GetRefError(
+        #[from]
+        #[backtrace]
+        GetRefError,
+    ),
+    #[error("Remote URL not found")]
+    NoRemoteUrl,
+}
```

Comment:
Currently your error is like a union of "everything that can go wrong under the hood" errors, but this is not very user friendly. For example, as a user I get `GetRefError` in response - how can I react to it, or what does it even tell me, really?

We consider errors a part of the API and design them NOT based on what can go wrong under the hood, but rather based on what logically can go wrong and how to communicate it best to the user ([this](https://github.com/kamu-data/kamu-cli/blob/e496fe036b58367a5119a1c25721d8c333ec65af/src/domain/core/src/entities/metadata_chain.rs#L614) is a good example).

I would start with including `Internal` error to represent cases where "some bad stuff happened under the hood". Then see what logical situations we should represent to make them distinguishable to the user. For example `GetRefError::NotFound` could be mapped into `RemoteDatasetNotFound` - so the user will know that remote dataset does not exist any more.

This alone might be enough for now - we can always come back and extract more logical errors later.

Reply:
done